### PR TITLE
improve dbiterator, accept 'SELECT' and 'FROM'

### PR DESCRIPTION
**Pull request by remicollet**
Just because this seems more legible:
```
      $sql = [
         'SELECT' => ['a', 'b'],
         'FROM'   => 'foo',
         'WHERE'  => ['c' => 1],
      ];
      $req = $DB->request($sql);
```

Also, use switch instead of if ... elseif... elseif...

### CODE REVIEW

File: inc/db.function.php [hunk 1]
```diff
@@ -1374,6 +1374,11 @@ function TableExists($tablename) {
 function FieldExists($table, $field, $usecache=true) {
    global $DB;
 
+   if (!TableExists($table)) {
+      trigger_error("Table $table does not exists", E_USER_WARNING);
```
Comment (trasher) [inline inc/db.function.php]: :+1:

File: inc/db.function.php [hunk 2]
```diff
@@ -1395,6 +1400,11 @@ function FieldExists($table, $field, $usecache=true) {
 function isIndex($table, $field) {
    global $DB;
 
+   if (!TableExists($table)) {
+      trigger_error("Table $table does not exists", E_USER_WARNING);
```
Comment (trasher) [inline inc/db.function.php]: :+1:

File: tests/SearchTest.php [hunk 3]
```diff
@@ -231,7 +236,7 @@ public function testSearchOptions() {
                    'users_id' => 0,
                    'num' => $key,
                );
-               $displaypref->add($input);
+                $displaypref->add($input);
```
Comment (trasher) [inline tests/SearchTest.php]: Isn't there a spacing issue here?

File: tests/bootstrap.php [hunk 4]
```diff
@@ -142,7 +155,7 @@ function loadDataset() {
    $CFG_GLPI['url_base']      = GLPI_URI;
    $CFG_GLPI['url_base_api']  = GLPI_URI . '/apirest.php';
 
-   @mkdir(GLPI_LOG_DIR, 0755, true);
+   is_dir(GLPI_LOG_DIR) or mkdir(GLPI_LOG_DIR, 0755, true);
```
Comment (trasher) [inline tests/bootstrap.php]: G-R-E-A-T :+1:  for the `@` removal :p

**Comment by remicollet** [comment]
@trashy, I use trigger_error, so, during **runtime**, these are manage by GLPI error handler as any other error (and traced in the php_errors.log).

Exception are for now, only for the test suite ;)

Yes... sometime... we'll have to switch to exceptions ;)

**Comment by trasher** [comment]
Oh, OK, I did not have in mind the specific error handler stuff... Thanks!

**Comment by remicollet** [comment]
Merged

**Pull request closed without merging**